windows/src/desktop/kmshell/main/Keyman.System.RemoteUpdateCheck.pas

```diff
@@ -375,4 +218,37 @@ function TRemoteUpdateCheck.DoRun: TRemoteUpdateCheckResult;
    KL.Log(LogMessage);
  end;
 
+function ConfigCheckContinue: Boolean;
+var
+  registry: TRegistryErrorControlled;
+begin
+{ Verify that it has been at least CheckPeriod days since last update check }
+  Result := False;
+  try
+    registry := TRegistryErrorControlled.Create;   // I2890
+    try
+      if registry.OpenKeyReadOnly(SRegKey_KeymanDesktop_CU) then
+      begin
+        if registry.ValueExists(SRegValue_CheckForUpdates) and not registry.ReadBool(SRegValue_CheckForUpdates) then
+        begin
+          Exit;
+        end;
+        if registry.ValueExists(SRegValue_LastUpdateCheckTime) and (Now - registry.ReadDateTime(SRegValue_LastUpdateCheckTime) > CheckPeriod) then
+        begin
+          Result := True;
+        end;
+      end;
+    finally
+      registry.Free;
+    end;
+  except
+    { we will not run the check if an error occurs reading the settings }
+    on E:Exception do
```

Comment:
It's best not to have 'catch-all' exception handlers: it's an antipattern. Restrict the handler to exceptions you expect, e.g. ERegistryException is it?